lib/builder.ts

```diff
@@ -309,7 +309,7 @@ class Builder extends EventEmitter {
     const tmpObj = tmp.dirSync({
       prefix: 'broccoli-',
       unsafeCleanup: true,
-      dir: this.tmpdir,
+      dir: this.tmpdir || undefined,
```

Comment:
Why is this needed? Do we have types for `this.tmpdir` wrong?

Reply:
the `dir` option only accepts `string | undefined` but `this.tmpdir` is `string | undefined | null`.

This casts `null` to `undefined` when passing it to `dirSync`.

I would have changed the type of `this.tmpdir` to not allow null, but it's technically a breaking change because the property is public.

Reply:
The type of `tmp.dirSync`'s `dir` is overly strict, not allowing `null`.

Reply:
I can adjust differently, but yes the types are somewhat misaglined

`string | null | undefined`  cannot be assigned to `string | undefined`

Reply:
(I suppose this also casts `""` to `undefined`, but an empty string seems bad in this case)

Reply:
Actually, I think this is a better option as is. Due to both keeping broccolis public types table, and observing the type preferences of `tmp.dirSync`